### DISCUSSION TITLE
Cannot navigate to a page not created by Nalu navigation

### DISCUSSION THREAD

**Discussion by CodingOctocat** [nalu-development/nalu#59] [Q&A] [answered]
This issue is an extension of the navigation mode suggestion you gave in #57. I've noticed a problem with the theme color of the **More Options** button on the navigation bar. It starts out normal black and then turns white. I can "refresh" to the normal color (black) by toggling the Tab. If the system theme is Dark mode, then there is no problem. The **More Options** button starts out the correct white color and does not change.

As you can see, at around 9s, **More Options** is normally black, but immediately switches to white.

![136](https://github.com/user-attachments/assets/98700dc7-db70-4536-b6a3-9b14e7dee894)
![139](https://github.com/user-attachments/assets/c1bf6578-e956-47b7-a4b4-32e8f8927519)

![动画](https://github.com/user-attachments/assets/e4dea1a3-b78f-411b-aec7-fb1984bec507)

---

I'm not sure if this is a NaluShell issue or a MAUI issue, but if the app's initial page is the AppShell and there is no secondary navigation, then this won't be an issue.

I tried to create a simple Demo to reproduce the issue, but ran into a new issue: **Cannot navigate to a page not created by Nalu navigation**

So, I have to fix that first.

[MoreOptionsThemeColorIssue.zip](https://github.com/user-attachments/files/18050651/MoreOptionsThemeColorIssue.zip)

**Comment by albyrock87** [comment]
Which version of MAUI are you using?

This issue you reported reminds me of these still opened PRs:

- https://github.com/dotnet/maui/pull/24766
- https://github.com/dotnet/maui/pull/25870
- https://github.com/dotnet/maui/pull/25102

Anyway, Nalu.Maui.Navigation has nothing to do with colors, so I strongly believe this issue is a MAUI one.

You can try to reproduce with a standard `Shell` and see if it happens or not, but I'm pretty sure it does.

**Reply by CodingOctocat** [reply]
I use .NET9(Microsoft.Maui.Controls 9.0.12)，I have tried to reproduce this using the standard `Shell` but it fails, if the initial page is not a `Shell` then I can't navigate through `Shell.Current.GotoAsync`.The problem now is that I failed to use Nalu as well, which is very strange, because in #57 I succeeded.

Also the code in [MoreOptionsThemeColorIssue.zip](https://github.com/user-attachments/files/18050651/MoreOptionsThemeColorIssue.zip) is based on the Nalu document. I don't know what's wrong with it. If I can fix it then I can test this issue further.

**Reply by albyrock87** [reply, marked as answer]
@CodingOctocat you forgot to remove all the `ContentTemplate="{DataTemplate ....}"` from `AppShell.xaml`.
`ContentTemplate` is only for the standard `Shell`, otherwise it conflicts with Nalu upon shell content page creation.

**Reply by CodingOctocat** [reply]
@albyrock87 Thanks, the app runs and the More Options icon colors are normal, MAUI always has the odd bug...
